### ISSUE TITLE
CS8618 (uninitialized field) when checking for null

### ISSUE THREAD

**Issue by Tragetaschen** [dotnet/roslyn#33391]
**Version Used**: 
16.0.0 Preview 3

**Steps to Reproduce**:

```csharp
class C
{
    readonly string s;
    C(bool a) // CS8618
    {
        if (a)
            s = "";
        if (s == null)
            throw new InvalidOperationException();
    }
}
```
This is similar to #25529.
After possibly setting the the field, the code checks if the field is still default-initialized and throws if not. After that, `s` is definitely non-null.

**Expected Behavior**:
The constructor either initializes or throws, so there shouldn't be a warning.

**Actual Behavior**:
CS8618 for the constructor

**Comment by Tragetaschen** [comment]
With a temporary variable in the constructor, the flow analysis works:
```csharp
class C
{
    readonly string s;
    C(bool a) // CS8618
    {
        string? sTmp = null;
        if (a)
            sTmp = "";
        s = sTmp ?? throw new InvalidOperationException();
    }
}
```

**Comment by RikkiGibson** [comment]
This will probably be resolved at the same time as #34598
